File: leet_345_reverse_vowels.cpp
class Solution {
public:
    string reverseVowels(string s)
    {
        vector<char>    c;
        vector<int>     idx;
        int             i = -1;

        while (++i < s.length())
        {
            if (isvowel(s[i]))
            {
                idx.push_back(i);
                c.push_back(s[i]);
            }
        }
        reverse(c.begin(), c.end());
        i = 0;
        for (int n : idx)
            s[n] = c[i++];
        return s;
    }

    int isvowel(char c)
    {
        if (c ^ 'a' && c ^ 'e' && c ^ 'i' && c ^ 'o' && c ^ 'u' &&
           c ^ 'A' && c ^ 'E' && c ^ 'I' && c ^ 'O' && c ^ 'U')
            return 0;
        return 1;
    }
};

File: leet_345_reverse_vowels.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        v = ['a','e','i','o','u','A','E','I','O','U']
        idx, c = [], []
        ss = list(s)
        # print(ss)
        for i in range(len(s)):
            if s[i] in v:
                idx.append(i)
                c.append(s[i])
        c = c[::-1]
        i = 0
        for n in idx:
            ss[n] = c[i]
            i += 1
        return ''.join(ss)
        
